Unix/tests/pal/test_pal.cpp

```diff
@@ -35,7 +35,10 @@
 #include <pal/file.h>
 #include <pal/hashmap.h>
 #include <pal/format.h>
+
+#if (!(defined(linux) || defined(macos)) && !defined(CONFIG_ENABLE_WCHAR))
```

Comment:
So it looks like this is coupled with similar `#ifdef` statement at line 2422?

You need explanation as to the problem, and perhaps changing `#ifdef` here to be coupled there.  Like perhaps:

```
#if (!(defined(linux) || defined(macos)) && !defined(CONFIG_ENABLE_WCHAR))
#include "test_pal_strings.h"
#else
#define WCHAR_TESTS_DISABLED
#endif
```

Then, at line 2422:

```
#if !defined(WCHAR_TESTS_DISABLED)
```

Then it's more obviouisly coupled.